From: Andrew Morton <akpm@linux-foundation.org>
Subject: revert "mm: align larger anonymous mappings on THP boundaries"
Date: Thu Dec  1 01:22:56 PM PST 2022

Several testers are seeing large slowdowns from this commit.  The reasons
for this are unclear.  Revert it from 6.1-rcX so this can be better
examined.

Fixes: f35b5d7d676e ("mm: align larger anonymous mappings on THP boundaries")
Reported-by: kernel test robot <yujie.liu@intel.com>
Reported-by: Nathan Chancellor <nathan@kernel.org>
Reported-by: Thorsten Leemhuis <regressions@leemhuis.info>
Acked-by: Yang Shi <shy828301@gmail.com>
Cc: Feng Tang <feng.tang@intel.com>
Cc: "Huang, Ying" <ying.huang@intel.com>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Rik van Riel <riel@surriel.com>
Cc: Xing Zhengjun <zhengjun.xing@linux.intel.com>
Cc: "Yin, Fengwei" <fengwei.yin@intel.com>
Cc: Yujie Liu <yujie.liu@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/mm/mmap.c~revert-mm-align-larger-anonymous-mappings-on-thp-boundaries
+++ a/mm/mmap.c
@@ -1779,9 +1779,6 @@ get_unmapped_area(struct file *file, uns
 		 */
 		pgoff = 0;
 		get_area = shmem_get_unmapped_area;
-	} else if (IS_ENABLED(CONFIG_TRANSPARENT_HUGEPAGE)) {
-		/* Ensures that larger anonymous mappings are THP aligned. */
-		get_area = thp_get_unmapped_area;
 	}
 
 	addr = get_area(file, addr, len, pgoff, flags);
_
